### ISSUE TITLE
Enhancement Request: fill_form functionality

### ISSUE THREAD

**Issue by shanemcquillan** [hellerbarde/stapler#19]
Hi. I'm wondering if you've given thought to, or considered adding the fill_form functionality of pdftk. I would be interested to know your thoughts on this.

**Comment by hellerbarde** [comment]
I haven't thought about it, because I never understood why it is useful. Can you explain what it is useful for?
And the second question is, does PyPDF support that functionality? If not, it won't be in stapler.

**Comment by hellerbarde** [comment]
OK. I checked, the functionality is provided in PyPDF2, so this is maybe possible.

**Comment by shanemcquillan** [comment]
You're correct. I actually found that and created a pull request for an extra function that will set these values across the entire document. Here's the forked version on my repo: https://github.com/mstamy2/PyPDF2
Should be very straight forward for you guys to add.

**Comment by shanemcquillan** [comment]
A feature like this would be useful for filling forms that require similar content regularly.

**Comment by clphillips** [comment]
I've got a number of projects relying on pdftk to programmatically fill out user supplied PDF forms with database driven data. It's actually the only reason I use pdftk, and probably the only reason I'd use stapler (if it had the feature).

So +1.

**Comment by hellerbarde** [comment]
Can you elaborate on that FDF file. Is that a common format? where can I find out more about it?

**Comment by clphillips** [comment]
Yes, FDF is a common format. Until recently it was the **only** format that could be used to fill forms for PDF documents. [More info here](http://www.verypdf.com/pdfform/fdf.htm), [official spec here](http://partners.adobe.com/public/developer/en/pdf/PDFReference16.pdf) (starts on page 671).

It's a simple file format, and is primarily used with the following flow:

1. Generate an FDF file from a PDF with form fields
2. Programmatically modify the FDF file to populate form data
3. Flatten the FDF and PDF into a new PDF document, which injects the form data and removes the form fields.
